docs/auth/jwt/jwt-configuration.mdx

```diff
@@ -0,0 +1,677 @@
+using various JWT issuers. This configuration option allows for greater flexibility in authentication.
+
+## Security considerations
+
+### Setting audience check
+
+Certain JWT providers share JWKs between multiple tenants (like Firebase). They use the `aud` claim of JWT to specify
+the intended tenant for the JWT. Setting the `audience` field in the Hasura JWT configuration will make sure that the
+`aud` claim from the JWT is also checked during verification. Not doing this check will allow JWTs issued for other
+tenants to be valid as well.
+
+In these cases, you **MUST** set the `audience` field to appropriate value. **Failing to do so is a major security
+vulnerability**.
+
+## JWT with the WebSocket protocol
+
+When executing a subscription (or query or mutation) over the WebSocket protocol, the authentication step is executed on
+`connection_init` when the websocket is connected to Hasura Engine and is valid until the expiry of the JWT when in JWT
+mode.
+
+Once authenticated, all operations are allowed without further check, until the authentication expires.
+
+## Popular providers and known issues
+
+### AWS Cognito
+
+AWS Cognito and ELB (Elastic Load Balancer) has a known issue where it adds additional padding (using = characters) to
+the JWT token that is generated from Cognito.
+
+This is a known issue and is documented by AWS in
+[their docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html#user-claims-encoding):
+
+> Standard libraries are not compatible with the padding that is included in the Application Load Balancer
+> authentication token in JWT format.
+
+Currently, there is no workaround possible in Hasura. Even if Hasura strips the additional padding the signature
+verification of the token would fail (as Hasura had to tamper the token).
+
+### Firebase
+
+Firebase publishes the JWKs at:
+
+[https://www.googleapis.com/service_accounts/v1/jwk/securetoken@system.gserviceaccount.com](https://www.googleapis.com/service_accounts/v1/jwk/securetoken@system.gserviceaccount.com)
+
+If you are using Firebase and Hasura, use this config:
+
+```json
+{
+  "jwk_url": "https://www.googleapis.com/service_accounts/v1/jwk/securetoken@system.gserviceaccount.com",
+  "audience": "<firebase-project-id>",
+  "issuer": "https://securetoken.google.com/<firebase-project-id>"
+}
+```
+
+### Auth0 {#auth0-issues}
+
+Refer to the [Auth0 JWT tutorial](/auth/jwt/tutorials/integrations/1-auth0.mdx) for a detailed guide on integrating
+Auth0 with Hasura.
+
+### Clerk
+
+Clerk integrates with Hasura GraphQL Engine using JWTs.
+
+Clerk publishes their JWK under: `https://<YOUR_CLERK_FRONTEND_API>/.well-known/jwks.json`
+
+Refer to the [Clerk JWT tutorial](/auth/jwt/tutorials/integrations/4-clerk.mdx) to set up authenticated requests to
+Hasura with Clerk.
+
+## Generate a JWT Config for Auth0 or Firebase {#generating-jwt-config}
```

Comment:
We should remove this section. The configuration syntax in DDN v3 is a bit different

Reply:
Done